libraries/ESP8266WiFi/src/BearSSLHelpers.h

```diff
@@ -123,6 +153,30 @@ class X509List {
     // Disable the copy constructor, we're pointer based
     X509List(const X509List& that) = delete;
 
+    // Allow moves
+    X509List(X509List&& that) {
+        _count = that._count;
+        _cert = that._cert;
+        _ta = that._ta;
+        that._count = 0;
+        that._cert = nullptr;
+        that._ta = nullptr;
```

Comment:
Now it's probably not a nit, but a real suggestion.
Introducing unique_ptr and also adding `member = $default_value$` we will remove a lot of boilerplate
(and not confuse the future reader of the purpose of these +100 lines)